File: frontend/lib/api/health.ts
import { fetchApi } from './client';

export interface HealthCheck {
  status: string;
  database?: string;
}

export const healthApi = {
  appHealth: () => fetchApi<HealthCheck>('/health'),
  dbHealth: () => fetchApi<HealthCheck>('/api/v1/database/health'),
};

File: backend/api/routes/health.py
from fastapi import APIRouter

router = APIRouter(tags=["Health & Service Info"])


@router.get("/health", summary="Health Check")
def health_check():
    """Returns service health status."""
    return {"status": "ok", "service": "FRADMS API"}


@router.get("/", summary="API Root Info")
def root_info():
    """Returns API service description and sitemap."""
    return {
        "service": "Fair Resource Allocation & Dynamic Matching System API",
        "version": "1.0.0",
        "documentation": "/docs",
        "redoc": "/redoc",
        "status": "healthy",
    }
